src/transformStateWithClones.js

```diff
@@ -8,6 +8,28 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const newState = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        currentState = {};
+        break;
+
+      case 'addProperties':
+        Object.assign(currentState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete currentState[key];
+        }
+    }
+    newState.push({ ...currentState });
```

Comment:
Pushing the copy of the current state to the `newState` array is correctly placed outside of the `switch` block, which is good practice as it avoids repetition.